### ISSUE TITLE
NuGet.exe push -Timeout parameter ignored

### ISSUE THREAD

**Issue by tbolon** [NuGet/Home#3631]
It seems to be a regression from previous versions. In latest 3.4.4 bits, the -Timeout parameter is not used, and the default values does not seems to match the command description.

See this log:

```
nuget.exe push "xxx.1.8.6127.17492.nupkg" -ApiKey xxx -Source xxx -Timeout 3599
  Pushing xxx.nupkg to 'xxx'...
    PUT ...
    PUT ...
    PUT ...
  The HTTP request to 'PUT ...' has timed out after 100000ms.
```

And the push help:

```
 -Timeout                      Specifies the timeout for pushing to a server in seconds. Defaults to 300 seconds (5 minutes).
```

**Comment by pberggreen** [comment]
It looks like a NuGet error:
https://github.com/NuGet/Home/issues/2785

**Comment by tbolon** [comment]
You are right, it is certainly solved in nuget 3.5
